tests/extra/test_stac_jobdb.py

```diff
@@ -0,0 +1,391 @@
+import datetime
+from unittest.mock import MagicMock, patch
+
+import geopandas as gpd
+import pandas as pd
+import pandas.testing as pdt
+import pystac
+import pystac_client
+import pytest
+from requests.auth import AuthBase
+from shapely.geometry import Point
+
+from openeo.extra.job_management import MultiBackendJobManager
+from openeo.extra.stac_job_db import STACAPIJobDatabase
+
+
+@pytest.fixture
+def mock_auth():
+    return MagicMock(spec=AuthBase)
+
+
+@pytest.fixture
+def mock_stac_api_job_database(mock_auth) -> STACAPIJobDatabase:
+    return STACAPIJobDatabase(collection_id="test_id", stac_root_url="http://fake-stac-api", auth=mock_auth)
```

Comment:
for dummy/fake  URLs, use the [`.test` TLD](https://en.wikipedia.org/wiki/.test), which is designed for especially for test situations